spark/agent/src/main/scala/za/co/absa/spline/harvester/builder/write/WriteCommandExtractor.scala

```diff
@@ -112,17 +110,29 @@ class WriteCommandExtractor(pathQualifier: PathQualifier, session: SparkSession)
     WriteCommand(name, sourceIdentifier, mode, query, Map("table" -> table))
   }
 
-  private val toBeImplemented = Seq(classOf[AlterTableAddColumnsCommand], classOf[AlterTableChangeColumnCommand],
-    classOf[AlterTableRenameCommand], classOf[AlterTableSetLocationCommand], classOf[CreateDataSourceTableCommand],
-    classOf[CreateDatabaseCommand], classOf[CreateTableLikeCommand], classOf[DropDatabaseCommand],
-    classOf[LoadDataCommand], classOf[TruncateTableCommand])
-
-  private def logUnimplementedComamnd(c: LogicalPlan): Unit = {
-    if (toBeImplemented.contains(c.getClass)) {
-      logger.warn(s"Spark command was intercepted, but is not yet implemented! Command:'${c.getClass}'")
+  private val commandsToBeImplemented = Seq(
+    classOf[AlterTableAddColumnsCommand],
+    classOf[AlterTableChangeColumnCommand],
+    classOf[AlterTableRenameCommand],
+    classOf[AlterTableSetLocationCommand],
+    classOf[CreateDataSourceTableCommand],
+    classOf[CreateDatabaseCommand],
+    classOf[CreateTableLikeCommand],
+    classOf[DropDatabaseCommand],
+    classOf[LoadDataCommand],
+    classOf[TruncateTableCommand]
+  )
+
+  private def alertWhenUnimplementedCommand(c: LogicalPlan): Unit = {
+    if (commandsToBeImplemented.contains(c.getClass)) {
+      val msg = s"Spark command was intercepted, but is not yet implemented! Command:'${c.getClass}'"
+
+      splineMode match {
+        case SplineMode.REQUIRED => throw new UnsupportedOperationException(msg)
```

Comment:
Can you please define a new checked exception class for this purpose and use that one instead?
Also please declare it in the method signature.
`UnsupportedOperationException` is too generic and might be thrown from underlying code in Spark or potentially other libraries. Also as it's not declared in the class contract the caller can easily miss it.